### PR TITLE
Add json-ld markup to news posts

### LINKED ISSUE #2457: Add json-ld markup to news posts
We have 'blogposts' https://training.galaxyproject.org/training-material/news.html which could use some json-ld markup to annotate metadata in a more easily consumable format for search engines.

Here are some examples of the appropriate JSON-LD https://jsonld.com/blog-post/

And this can be added to https://github.com/galaxyproject/training-material/blob/main/_layouts/news.html

### PR DESCRIPTION
**Pull request by AfiDede**
Fixes #2457
@hexylena Kindly let me know what you think.
I am new to jekyll and still learning around it. Do let me know what I can do to make it better if this cannot be merged.
Thanks.

### CODE REVIEW

File: _layouts/news.html [hunk 1]
```diff
@@ -19,7 +19,48 @@
  {% assign coverimage = "/assets/images/GTN.png" %}
  {% assign coverimagealt = "GTN logo with a multi-coloured star and the words Galaxy Training Network" %}
 {% endif %}
-
+<script type="application/ld+json">
+    {
+      "@context": "https://schema.org",
+      "@type": "BlogPosting",
+      "url": "{{ site.url }}{{ page.url }}",
```
Comment (hexylena) [inline _layouts/news.html:26]:
```suggestion
      "url": "{{ site.url }}{{ site.baseurl }}{{ page.url }}",
```

File: _layouts/news.html [hunk 2]
```diff
@@ -19,7 +19,48 @@
...
+      },
+      "publisher": {
+        "@type": "Organization",
+        "name": {{ site.title | jsonify }},
+        "url": "{{ site.url }}",
+        "logo": {
+          "@type": "ImageObject",
+          "width": 32,
+          "height": 32,
+          "url": "{{ site.url }}/icon/favicon.ico"
```
Comment (hexylena) [inline _layouts/news.html:49]:
```suggestion
          "url": "{{ site.url }}{{ site.baseurl }}{{ site.small_logo }}"
```

that logo is 60x60 px

File: _layouts/news.html [hunk 3]
```diff
@@ -19,7 +19,48 @@
...
+        }
+      },
+      "mainEntityOfPage": {
+        "@type": "WebPage",
+        "@id": "{{ site.url }}{{ page.url }}"
```
Comment (hexylena) [inline _layouts/news.html:54]:
```suggestion
        "@id": "{{ site.url }}{{ site.baseurl }}{{ page.url }}"
```

File: _layouts/news.html [hunk 4]
```diff
@@ -19,7 +19,48 @@
...
+      },
+      "image": {
+        "@type": "ImageObject",
+        "width": {{ page.img_width | default: site.img_width }},
+        "height": {{ page.img_height | default: site.img_height }},
+        "url": "{{ site.url }}{{ page.img_url | default: site.img_url }}"
```
Comment (hexylena) [inline _layouts/news.html:60]: The news posts use `page.cover` for their cover image.  I guess default to the small_logo like above, if it isn't set.

**Review by hexylena — COMMENTED** [review]
I've run this locally and I currently see the following which looks like a very good start, but some things we can fix which I've commented on in the PR. Exciting!

```
<script type="application/ld+json">
    {
      "@context": "https://schema.org",
      "@type": "BlogPosting",
      "url": "http://localhost:4002/news/2021/10/12/speed.html",
      "name": "Attention Contributors: GTN Performance Enhancements",
      "headline": "Attention Contributors: GTN Performance Enhancements",
      "keywords": "authors,developers,performance",
      "description": "<p>Are you one of our amazing GTN Contributors? Were you ever bothered by how slow it was to build the site locally? This is now fixed completely! We have replaced one of the slowest steps of the GTN’s build process and rewritten it ourselves to be more performant.</p>",
      "articleBody": "Are you one of our amazing GTN Contributors? Were you ever bothered by how slow it was to build the site locally? This is now fixed completely! We have replaced one of the slowest steps of the GTN’s build process and rewritten it ourselves to be more performant.\n\nAdditionally this means when you make contributions to the GTN you no longer need to wait 20 minutes to have all of the links checked in your contribution before it can be merged, as that step now completes in under 1 minute!\n\nHappy contributing!\n",
      "datePublished": "2021-10-12 00:00:00 +0200",
      "dateModified": "2021-10-12 10:52:35 +0200",
      "author": {
        "@type": "Person",
        "name": null,
        "givenName": null,
        "familyName": null,
        "email": "galaxytrainingnetwork@gmail.com"
      },
      "publisher": {
        "@type": "Organization",
        "name": "Galaxy Training!",
        "url": "http://localhost:4002",
        "logo": {
          "@type": "ImageObject",
          "width": 32,
          "height": 32,
          "url": "http://localhost:4002/icon/favicon.ico"
        }
      },
      "mainEntityOfPage": {
        "@type": "WebPage",
        "@id": "http://localhost:4002/news/2021/10/12/speed.html"
      },
      "image": {
        "@type": "ImageObject",
        "width": ,
        "height": ,
        "url": "http://localhost:4002"
      }
    }
    </script>
```

File: _layouts/news.html (hunk 4 above)
Comment (AfiDede) [reply _layouts/news.html:60]: Thank you @hexylena I am working on these right away !

Comment (AfiDede) [reply _layouts/news.html]: Hi @hexylena  I have updated and made the necessary changes now.

**Review by hexylena — APPROVED** [review]

**Comment by hexylena** [comment]
I'm sorry for the long long delay @AfiDede but we finally merge this! Thank you so much for the contribution and working out the metadata

**Pull request merged**